Remove ``__UTC__`` crutch for Python 3.10 support

tbowers7 opened the pull request:
When ``datetime.datetime.utcnow()`` was deprecated in Python 3.12, we needed some crutches (#1785) to be able to successfully transition to the new ``datetime.UTC`` constant (introduced in Python 3.11).  Since PypeIt no longer supports Python 3.10 (#1854), we can safely remove these crutches to provide for cleaner code relying solely on the ``datetime.UTC`` constant.

DevSuite tests pass (Nautilus report ``tbowers-py310-sep19.report``):

```
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  261 passed, 443 warnings in 508.66s (0:08:28) ---
--- PYTEST UNIT TESTS PASSED  157 passed, 93167 warnings in 959.23s (0:15:59) ---
--- PYTEST VET TESTS PASSED  69 passed, 135631 warnings in 4766.40s (1:19:26) ---
--- PYPEIT DEVELOPMENT SUITE PASSED 280/280 TESTS  ---
Testing Started at 2025-09-19T17:10:25.540237
Testing Completed at 2025-09-20T11:39:28.966919
Total Time: 18:29:03.426682
```

tbowers7 committed:
Remove ``__UTC__`` python <= 3.10 compatability

Since we now only support Python >= 3.11, remove these crutches left
over from supporting 3.10.

Also, clean up import order somewhat for sanity.

	modified:   pypeit/calibrations.py
	modified:   pypeit/inputfiles.py
	modified:   pypeit/pypeit.py
	modified:   pypeit/pypmsgs.py
	modified:   pypeit/scripts/chk_for_calibs.py
	modified:   pypeit/setup_gui/controller.py

tbowers7 committed:
Release notes

	modified:   doc/releases/1.18.1dev.rst

kbwestfall commented on pypeit/calibrations.py:
```diff
@@ -233,7 +221,7 @@ def check_calibrations(self, file_list, check_lamps=True):
         for ii, ifile in enumerate(file_list):
             # Save the lamp status
-            headarr = deepcopy(self.spectrograph.get_headarr(ifile))
+            headarr = copy.deepcopy(self.spectrograph.get_headarr(ifile))
```
For these header copies, I think we should be using the class method explicitly: https://docs.astropy.org/en/latest/io/fits/api/headers.html#astropy.io.fits.Header.copy (even though the above line of code does this implicitly).

tepickering replied:
in the interest of being explicit and clear, i am inclined to agree, but don't feel super strongly about it.

tbowers7 replied:
Now reads:
```
headarr = [h.copy() for h in self.spectrograph.get_headarr(ifile)]
```

kbwestfall reviewed (approved):
I had one minor comment about using the copy class method for headers, but otherwise this looks good to go.  Thanks!

tepickering reviewed (approved):
thanks for this long-overdue cleanup! looks good and thanks also for some of the import cleanups.

bpholden reviewed (approved):
Looks good according to the other Tim

tbowers7 committed:
Explicitly use the ``Header.copy()`` method

In calibrations.py, explicitly use the ``astropy.io.fits.Header.copy()``
method instead of the ``copy.deepcopy`` abstraction layer, which calls
the .copy() method anyway.

	modified:   pypeit/calibrations.py
